src/driver.ml

```diff
@@ -1315,6 +1279,33 @@ let standalone () =
     Caml.exit 1
 ;;
 
+let run_as_ppx_rewriter () =
+  let usage = Printf.sprintf "%s [extra_args] <infile> <outfile>" exe_name in
+  let valid_args = List.rev !args in
+  let input = Caml.Sys.argv in
+  match List.rev @@ Array.to_list @@ input with
+  | output_fn :: input_fn :: flags_and_prog_name
+    when List.length flags_and_prog_name > 0 -> (
+      let prog_name_and_flags = List.rev flags_and_prog_name |> Array.of_list in
+      parse_input prog_name_and_flags ~valid_args ~incorrect_input_msg:usage;
+      perform_checks := false;
+      interpret_mask ();
+      try
+        rewrite_binary_ast_file input_fn output_fn;
+        Caml.exit 0
+      with exn ->
+        Location.report_exception Caml.Format.err_formatter exn;
+        Caml.exit 1 )
+  | [ help; _ ] when String.equal help "-help" || String.equal help "--help" ->
+      parse_input input ~valid_args ~incorrect_input_msg:usage;
+      (* The program should already have terminated with 0 *)
+      Printf.eprintf "Usage: %s\n%!" usage;
```

Comment:
We should make it explicit that we expect this to be unreachable by using `assert false`

Reply:
Yes, totally. Thanks!